apps/files/src/store/actions.js

```diff
@@ -283,7 +283,7 @@ function _buildCollaboratorShare (s, file) {
 
   // expiration:Object if unset, or string "2019-04-24 00:00:00"
   if (typeof s.expiration === 'string' || s.expiration instanceof String) {
-    share.expires = Date.parse(s.expiration)
+    share.expires = new Date(s.expiration)
```

Comment:
add `// FIXME: use moment`
parsing the date like this is likely to cause troubles with timezones

Reply:
With Date object, you get local timezone that is set in the OS of the user. This is an example of what date returns - `Sat Feb 29 2020 00:00:00 GMT+0100 (Central European Standard Time)`

I think it would be better not to use moment directly here because than we are enforcing it on all later use cases.

Reply:
Ah ok. So if "s.expiration" contains timezone info, then using `new Date()` is fine. I thought it was a milliseconds value in which case it would use the local timezone.